### DISCUSSION TITLE
Nextjs hydratation error

### DISCUSSION THREAD

**Discussion by electroheadfx** [47ng/nuqs#546] [Q&A] [answered]
When I re-render the page, I have hydratation error, How I can init default nuqs param with SSR/page mode ?
thanks

**Comment by franky47** [comment]
Could you give us more context about your setup please? (`nuqs` version, Next.js version, pages/app directory, page rendering mode).

From your comment, I assume you are using the Pages router in Next.js, in a statically optimised page.

If so, start with https://github.com/47ng/nuqs/issues/290 and see if isolating your searchParams-enabled components from the build render helps. Otherwise, for a true SSR (at request time) page, you'd need to use `getServerSideProps`, which guarantees the SSR render to have access to the correct search params.

**Comment by electroheadfx** [comment, marked as answer]
you are right, it works now with a getServerSideProps, I forgot it, thanks !
